modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -897,6 +905,7 @@ public CompletableFuture<Ignite> start(Path configPath) {
                     .whenCompleteAsync((res, ex) -> {
                         startupExecutor.shutdownNow();
                     });
+            return CompletableFuture.allOf(startFuturesBeforeJoin.toArray(CompletableFuture[]::new)).thenCompose((ignored) -> startFuture);
```

Comment:
Why do we need `thenCompose` here, can we pass the `startFuture` to `allOf` instead?

Reply:
refactored